File: 0046-permutations/0046-permutations.cpp
class Solution {
public:
    void solve(int index, vector<int>& nums,vector<int>& temp,vector<vector<int>>& finalAns) {
        if (index==nums.size()) {
            finalAns.push_back(temp);
            return;
        }
        //For index position, try all possible options
        for (int i=0;i<nums.size();i++) {
            if (nums[i]!=11) {
                int tempVar = nums[i];
                temp.push_back(nums[i]);
                nums[i]=11;

                solve(index+1,nums,temp,finalAns);
                temp.pop_back();
                nums[i] = tempVar;
            }
        }
    }
    vector<vector<int>> permute(vector<int>& nums) {
        vector<vector<int>> finalAns;
        vector<int> temp;
         solve(0,nums,temp,finalAns);
         return finalAns;
        
        
    }
};

File: 0046-permutations/0046-permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        finalAns = []
        def solve(index,temp):
            if index==len(nums): 
                finalAns.append(list(temp))
                return
            for i in range(len(nums)):
                if nums[i]!=11:
                    tempVar = nums[i]
                    temp.append(nums[i])
                    nums[i]=11
                    solve(index+1,temp)
                    temp.pop()
                    nums[i] = tempVar
        solve(0,[])
        return finalAns
        